test/letters/containers/AddressSection.unit.spec.jsx

```diff
@@ -272,47 +268,57 @@ describe('<AddressSection>', () => {
       stateCode: '',
       type: ADDRESS_TYPES.domestic
     };
-    const tree = SkinDeep.shallowRender(<AddressSection {...props}/>);
+    const tree = shallow(<AddressSection {...props}/>);
 
-    const instance = tree.getMountedInstance();
-    expect(instance.state.isEditingAddress).to.be.true;
+    expect(tree.state('isEditingAddress')).to.be.true;
   });
 
   describe('validation', () => {
     // Spy on all the validation functions!
     beforeEach(spyOnValidators);
 
-    // Clean up the spies so nobody finds out
+    // Extract the spies so nobody finds out
     afterEach(cleanUpSpies);
 
     it('should run all validations against a field if the address is valid', () => {
-      const tree = SkinDeep.shallowRender(<AddressSection {...defaultProps}/>);
-      const instance = tree.getMountedInstance();
+      const tree = mount(<AddressSection {...defaultProps}/>);
+
+      // Start editing
+      tree.find('button.usa-button-outline').simulate('click');
+
+      // Change the city and blur for the validation to run
+      tree.find('input[name="city"]').simulate('change', { target: { value: 'Elsweyre' } });
+      tree.find('input[name="city"]').simulate('blur');
 
-      instance.handleChange('city', 'Elsweyre');
-      instance.dirtyInput('city');
       expect(AddressSection.fieldValidations.city.every(validator => validator.called)).to.be.true;
     });
 
     it('should return the first error message it finds', () => {
-      const tree = SkinDeep.shallowRender(<AddressSection {...defaultProps}/>);
-      const instance = tree.getMountedInstance();
+      const tree = mount(<AddressSection {...defaultProps}/>);
+
+      // Start editing
+      tree.find('button.usa-button-outline').simulate('click');
+
+      // Change the city and blur for the validation to run
+      tree.find('input[name="city"]').simulate('change', { target: { value: '' } });
+      tree.find('input[name="city"]').simulate('blur');
 
-      instance.handleChange('city', '');
-      instance.dirtyInput('city');
       // The required validator (first in the list) should return an error message and no other validators should run
       expect(AddressSection.fieldValidations.city[0].called).to.be.true;
       expect(AddressSection.fieldValidations.city.slice(1).every(validator => !validator.called)).to.be.true;
     });
 
     it('should run validations on modified fields only', () => {
-      const tree = SkinDeep.shallowRender(<AddressSection {...defaultProps}/>);
-      const instance = tree.getMountedInstance();
+      const tree = mount(<AddressSection {...defaultProps}/>);
 
-      const fieldsToModify = ['city', 'stateCode'];
+      // Start editing
+      tree.find('button.usa-button-outline').simulate('click');
+
+      const fieldsToModify = ['city', 'addressOne'];
       fieldsToModify.forEach(field => {
-        instance.handleChange(field, '');
-        instance.dirtyInput(field);
+        const input = tree.find(`input[name="${field}"]`);
```

Comment:
Before I changed from modifying the `state` field, I found I couldn't look for either `input`s _or_ `select`s by doing
```js
tree.find(`input[name="${field}"], select[name="${field}"]`)
```
which is kind of bothersome.

Reply:
So only one selector at a time? Interesting. Also looks like you can search on object properties, though not sure that would be possible here. http://airbnb.io/enzyme/docs/api/selector.html#4-object-property-selector

Reply:
Yeah, that's the `[name="${field}"]` bit, but if I just search for that, it _also_ grabs the related `<label>`s, which just makes things a bit more verbose.